rllib/algorithms/dt/tests/test_dt_model.py

```diff
@@ -0,0 +1,241 @@
+import unittest
+
+import gym
+import numpy as np
+
+import ray
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+from rllib.algorithms.dt.dt_torch_model import DTTorchModel
+
+tf1, tf, tfv = try_import_tf()
+torch, _ = try_import_torch()
+
+
+def _assert_outputs_equal(outputs):
+    for i in range(1, len(outputs)):
+        for key in outputs[0].keys():
+            assert np.allclose(outputs[0][key], outputs[i][key])
+
+
+def _assert_outputs_not_equal(outputs):
+    for i in range(1, len(outputs)):
+        for key in outputs[0].keys():
+            assert not np.allclose(outputs[0][key], outputs[i][key])
+
+
+class TestDTModel(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        ray.shutdown()
+
+    def test_torch_model_init(self):
+        """Test models are initialized properly"""
+        model_config = {
+            "embed_dim": 32,
+            "num_layers": 2,
+            "max_seq_len": 4,
+            "max_ep_len": 10,
+            "num_heads": 2,
+            "embed_pdrop": 0.1,
+            "resid_pdrop": 0.1,
+            "attn_pdrop": 0.1,
+            "use_obs_output": False,
+            "use_return_output": False,
+        }
+
+        num_outputs = 2
+        observation_space = gym.spaces.Box(-1.0, 1.0, shape=(num_outputs,))
+
+        action_dim = 5
+        action_spaces = [
+            gym.spaces.Box(-1.0, 1.0, shape=(action_dim,)),
+            gym.spaces.Discrete(action_dim),
+        ]
+
+        B, T = 3, 4
+
```

Comment:
**TODO** maybe create the shared stuff inside setupclass method to avoid code repetition.

Reply:
I am actually changing the model_config just to make sure i didn't do any copypaste error where I used a constant. But I did move input_dict to its own thing

Reply:
no I meant the sample_batch that you create in each case.

Reply:
yes that's what I did, i'm not sure why it's not showing up when clicking "view changes", but if you go to the lastest commit you'll see the change in files